docs/swagger/spec.yml

```diff
@@ -611,38 +655,225 @@ paths:
+    delete:
+      parameters:
+        - name: refreshTokenId
+          in: path
+          required: true
+          type: string
+      summary: Delete a refresh token and revoke its access
+      tags:
+        - Authentication
+      responses:
+        204:
+          description: Revocation successful, No Content
+  /models/:
+    get:
+      summary: Lists models that perform a set of geoprocessing/map algebra functions
+      tags:
+        - Model Lab
       parameters:
         - $ref: '#/parameters/orderingBase'
         - $ref: '#/parameters/pageSize'
         - $ref: '#/parameters/page'
-        - $ref: '#/parameters/scene'
-        - $ref: '#/parameters/organization'          
+        - $ref: '#/parameters/minRating'
+        - $ref: '#/parameters/maxRating'
+        - $ref: '#/parameters/modelCategory'
+        - $ref: '#/parameters/modelTag'
+        - $ref: '#/parameters/search'
       responses:
         200:
-          description: Paginated list of boundaries
+          description: Returns a paginated list of models
           schema:
-            $ref: '#/definitions/BoundaryPaginated'
-  /boundaries/{uuid}/:
+            $ref: '#/definitions/ModelPaginated'
+        404:
+          description: Error message if insufficient permissions or objects not present
+    post:
+      summary: Create a new geoprocessing model
+      tags:
+        - Model Lab
+      responses:
+        201:
+          description: Create a new geoprocessing model
+          schema:
+            $ref: '#/definitions/Model'
+        default:
+          description: Error message if insufficient permissions to create a new model
+  /models/{uuid}/:
     get:
-      summary: Boundary details
+      summary: Retrieve details for a single geoprocessing model
       tags:
-        - Imagery
+        - Model Lab
       parameters:
         - $ref: '#/parameters/uuid'
       responses:
         200:
-          description: Details of Boundary object
+          description: Geoprocessing model record
           schema:
-            $ref: '#/definitions/Boundary'
+            $ref: '#/definitions/Model'
         404:
-          description: Boundary UUID was not found
+          description: Error message returned if insufficent permissions or model does not exist
+    delete:
+      summary: Delete a model
+      tags:
+        - Model Lab
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        204:
+          description: Delete successful
+        default:
+          description: Delete unsuccessful, either object did not exist or insufficient permissions
+    put:
+      summary: Update a geoprocessing model
+      tags:
+        - Model Lab
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        204:
+          description: Update an existing geoprocessing model
+        404:
+          description: Error message if insufficient permissions to update a model
+  /model-tags/:
+    get:
```

Comment:
We probably want to have some filters defined for this.
This endpoint will also likely be used to populate an autocomplete field in the future, so a fuzzy filter of some sort would make sense.